src/api/java/com/minecolonies/api/util/BlockStateUtils.java

```diff
@@ -0,0 +1,218 @@
+        {
+            return state1.getValue(propertyOne) == state2.getValue(propertyOne);
+        }
+
+        final IProperty propertyTwo = getPropertyByNameFromState(state2, propertyName);
+
+        if (propertyOne != null && propertyTwo != null && state1.getPropertyKeys().contains(propertyOne) && state2.getPropertyKeys().contains(propertyTwo))
+        {
+            return state1.getValue(propertyOne).toString().equals((state2.getValue(propertyTwo)).toString());
+        }
+        return false;
+    }
+
+    /**
+     * Get the property object of a state matching the given name
+     * Caches lookups in the propertyBlockMap hashmap
+     *
+     * @param state Blockstate we're checking for a property
+     * @param name  name of the property to find
+     */
+    public static IProperty getPropertyByNameFromState(@NotNull final IBlockState state, @NotNull final String name)
+    {
+        IProperty property = propertyBlockMap.get(state.getBlock().getClass().getName() + name);
+
+        if (property != null && state.getPropertyKeys().contains(property))
+        {
+            return property;
+        }
+        else
+        {
+            // Cached map entry nonexistant or wrong, calculate new
+            property = getPropertyByName(state.getPropertyKeys(), name);
+
+            if (property != null)
+            {
+                propertyBlockMap.put(state.getBlock().getClass().getName() + name, property);
+            }
+            return property;
+        }
+    }
+
+    /**
+     * Checks a list of properties for a matching name.
+     *
+     * @param properties the properties to check
+     * @param name       the property name we're looking for
+     * @return IProperty object found
+     */
+    public static IProperty getPropertyByName(@NotNull final Collection<IProperty<?>> properties, @NotNull final String name)
+    {
+        for (final IProperty tProperty : properties)
+        {
+            if (tProperty.getName().equals(name))
+            {
+                return tProperty;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns a copy of the state without the given property
+     *
+     * @param state    The state to use
+     * @param property The property to remove
+     * @return State without property
+     */
+    public static IBlockState getStateWithoutProperty(@NotNull final IBlockState state, final IProperty property)
+    {
+        final Collection<IProperty<?>> tList = state.getPropertyKeys();
+
+        for (final IProperty prop : state.getPropertyKeys())
+        {
+            if (prop == property)
+            {
+                tList.remove(prop);
+                break;
+            }
+        }
+
+        final IProperty[] tempArray = new IProperty[tList.size()];
+        tList.toArray(tempArray);
+
+        IBlockState newState = new BlockStateContainer(state.getBlock(), tempArray).getBaseState();
```

Comment:
getDefaultState does contain the properties though, just with the properties default value. Can't remove a property that way

Reply:
The problem is that you have literally no idea what blockstate type you get.
In general on the server you only get BlockStateContainer.

But depending on the Block you can also get Forges ExtendedBlockStateContainer, with its unlisted properties, which you do not even take into account here.

What ever. I am opposed to the way you have done this. But @Raycoms is responsible for that part of the code so if he :shipit: then I am fine with that.

Reply:
hm you got a valid point there, with other block state types it can't really fulfill its intended functionality. I think I'll just remove it then (not used anywhere anyways)